spring-boot/proxy/src/main/java/eu/coatrack/proxy/filters/pre/ApiKeyAuthFilter.java

```diff
@@ -43,25 +43,30 @@
  *
  * @author gr-hovest
  */
-public class ApiKeyAuthFilter extends GenericFilterBean {
+public class ApiKeyAuthFilter extends OncePerRequestFilter {
 
     private static final Logger log = LoggerFactory.getLogger(ApiKeyAuthFilter.class);
 
     @Autowired
     private ApiKeyAuthTokenVerifier apiKeyAuthTokenVerifier;
 
     @Override
-    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
-        log.debug(String.format("Api key auth filter received request of type %s", servletRequest.getClass()));
+    protected boolean shouldNotFilter(HttpServletRequest request) throws ServletException {
+        AntPathMatcher pathMatcher = new AntPathMatcher();
+        return pathMatcher.match("/favicon.ico", request.getServletPath());
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest servletRequest, HttpServletResponse servletResponse, FilterChain filterChain) throws ServletException, IOException {
```

Comment:
The function signature is longer than the maximum length a line of code should have. ~120 chars should not be exceeded. I suggest to make a line break before FilterChain.

Reply:
Done